tests/test_resources_job.py

```diff
@@ -131,15 +132,35 @@ def test_extra_vars_at_runtime(self):
             with mock.patch.object(click, 'edit') as edit:
                 edit.return_value = '# Nothing.\nfoo: bar'
                 result = self.res.launch(1, no_input=False)
-                self.assertTrue(
-                    edit.mock_calls[0][1][0].endswith('spam: eggs'),
+                self.assertDictContainsSubset(
+                    {"spam": "eggs"},
+                    yaml.load(edit.mock_calls[0][1][0])
                 )
-            self.assertEqual(
-                json.loads(t.requests[3].body)['extra_vars'],
-                'foo: bar',
+            self.assertDictContainsSubset(
+                {'foo': 'bar'},
+                json.loads(json.loads(t.requests[3].body)['extra_vars'])
             )
             self.assertDictContainsSubset({'changed': True, 'id': 42}, result)
 
+            # test with YAML and comments
+            t.register_json('/job_templates/1/', {
+                'ask_variables_on_launch': True,
+                'extra_vars': 'spam: eggs\n# comment',
+                'id': 1,
+                'name': 'frobnicate',
+                'related': {'launch': '/job_templates/1/launch/'},
+            })
+            with mock.patch.object(click, 'edit') as edit:
+                edit.return_value = '# Nothing.\nfoo: bar'
+                result = self.res.launch(
+                    1, no_input=False
+                )
+                self.assertIn('# comment', edit.mock_calls[0][1][0])
```

Comment:
This tests to make sure that if the Job Template extra_vars contain comments, those are displayed to the user when prompting for extra_vars. The Job Template extra_vars could be JSON, so that's a different test case.